google_analytics/ga_homepage_report.py

```diff
@@ -1,365 +1,534 @@
-percent_fmt = workbook.add_format({'num_format': '0.0%', 'bold': False})
-comma_fmt = workbook.add_format({'num_format': '#,##0', 'bold': False})
-date_fmt = workbook.add_format({'num_format': 'dd/mm/yy'})
-cell_format = workbook.add_format({'bold': True, 'italic': False})
-
-worksheet.conditional_format('F1:F1000', {'type': '3_color_scale'})
-worksheet.conditional_format('D1:D1000', {'type': '3_color_scale'})
-worksheet.conditional_format('C1:C1000', {'type': '3_color_scale'})
-worksheet.conditional_format('G1:G1000', {'type': '3_color_scale'})
-
-worksheet.set_column('A:A', 60)
-worksheet.set_column('B:B', 15, comma_fmt)
-worksheet.set_column('E:E', 15, comma_fmt)
-worksheet.set_column('C:D', 15, percent_fmt)
-worksheet.set_column('F:G', 15, percent_fmt)
-worksheet.write('A1', 'Homepage Course Enrollments, Data from '+str(d1)+' to '+str(d2) , cell_format)
-worksheet.write('A3', 'Total Homepage Views:', cell_format)
-worksheet.write('A4', 'Total Click-Throughs to Courses Below:', cell_format)
-worksheet.write('A5', 'Total CTR:', cell_format)
-worksheet.write('A6', 'Total Enrollments:', cell_format)
-worksheet.write('A7', 'Total Enrollments per Homepage View:', cell_format)
-
-worksheet.write('B3', hpTotal, comma_fmt)
-worksheet.write('B4', int(final12['uniqueViews'].sum()), comma_fmt)
-worksheet.write('B5', int(final12['uniqueViews'].sum())/hpTotal, percent_fmt)
-worksheet.write('B6', int(final12['uniqueEnrolls'].sum()), comma_fmt)
-worksheet.write('B7', int(final12['uniqueEnrolls'].sum())/hpTotal, percent_fmt)
-
-
-worksheet = writer.sheets['HomepageSubjects']
-
-worksheet.conditional_format('C1:C1000', {'type': '3_color_scale'})
-
-worksheet.set_column('A:A', 27)
-worksheet.set_column('B:B', 15, comma_fmt)
-worksheet.set_column('C:S', 15, percent_fmt)
-worksheet.write('A1', 'Top Subject Pages from the Homepage, Data from '+str(d1)+' to '+str(d2) , cell_format)
-
-worksheet = writer.sheets['CourseEnrolls']
-
-worksheet.conditional_format('B1:B1000', {'type': '3_color_scale'})
+from apiclient.discovery import build
+from oauth2client import client, file, tools
+import pandas
+from pandas import DataFrame, ExcelWriter
+import xlsxwriter
 
-worksheet.set_column('A:A', 100)
-worksheet.set_column('B:B', 20, comma_fmt)
-worksheet.write('A1', 'Course Enrollments from the Homepage, Data from '+str(d1)+' to '+str(d2) , cell_format)
-worksheet.autofilter('A3:B3')
-
-writer.save()
+################################################################
+#
+# Run this script by providing the following inputs
+#     python ga_homepage_report.py yyyy-mm-dd yyyy-mm-dd /path/to/report/destination/
+#
+# The first argument is the start date
+# The second argument is the end date
+# The third argument is the location the report will be written to
+#
+###############################################################
+
+def get_service():
+    """Get a service that communicates to a Google API."""
+    api_name = 'analytics'
```

Comment:
They are only ever used within this method.  I thought it more readable to keep them within the method